### PR TITLE
Remove process when its done

### LINKED ISSUE #653: Processes are not removed when completed
This is relatively minor as it doesn't cause any problems.

Currently after a protocols is completed it is not removed from the state, and neither are its adjudicator event subscriptions. It'd be nice if these were cleaned up on completion. It currently causes warnings in the console as adjudicator events are still dispatched to the completed process.

### PR DESCRIPTION
**Pull request by lalexgap**
Fixes #653.

Removes processes when they're done.

### CODE REVIEW

File: packages/wallet/src/redux/state.ts [hunk 1]
```diff
@@ -322,3 +335,13 @@ export function getCommitments(store: SharedData, channelId: string): Commitment
...
 
 export { NewLedgerChannel };
+
+export function isTerminalProcessProtocolState(
+  protocolState: ProtocolState,
+): protocolState is TerminalApplicationState | TerminalFundingState | TerminalDefundingState {
+  return (
+    (isApplicationState(protocolState) && isTerminalApplicationState(protocolState)) ||
+    (isFundingState(protocolState) && isTerminalFundingState(protocolState)) ||
+    (isDefundingState(protocolState) && isTerminalDefundingState(protocolState))
```
Comment (geoknee) [inline packages/wallet/src/redux/state.ts]: What about the `TerminalConcludingState` case?

File: packages/wallet/src/redux/state.ts [hunk 2]
```diff
@@ -322,3 +335,13 @@ export function getCommitments(store: SharedData, channelId: string): Commitment
 }
 
 export { NewLedgerChannel };
+
+export function isTerminalProcessProtocolState(
```
Comment (geoknee) [inline packages/wallet/src/redux/state.ts]: Is `isTerminalProtocolState` a better name?

File: packages/wallet/src/redux/state.ts (hunk 1 above)
Comment (lalexgap) [reply packages/wallet/src/redux/state.ts]: Oops, fixed!

File: packages/wallet/src/redux/state.ts (hunk 2 above)
Comment (lalexgap) [reply packages/wallet/src/redux/state.ts]: Renamed

File: packages/wallet/src/redux/__tests__/initialized.test.ts [hunk 3]
```diff
@@ -91,3 +91,34 @@ describe('when a updateSharedData action arrives', () => {
...
+    protocolState: { type: 'Funding.Success' },
+    sharedData: 'sharedData ',
+  }));
+  Object.defineProperty(fundProtocol, 'fundingReducer', {
+    value: reducer,
+  });
+
+  const result = walletReducer(state, action);
+  it('removes the current process id', () => {
+    expect(result.currentProcessId).toBeUndefined();
```
[the remark this answers is not in the document]
Comment (lalexgap) [reply packages/wallet/src/redux/__tests__/initialized.test.ts]: I think we'll need to think further about handling concurrent running processes so when we make those changes we can probably change this.

[the remark this answers is not in the document]
Comment (lalexgap) [reply packages/wallet/src/redux/__tests__/initialized.test.ts]: I've created #669 to address handling concurrent processes.

[the remark this answers is not in the document]
Comment (geoknee) [reply packages/wallet/src/redux/__tests__/initialized.test.ts]: 👍

**Pull request merged**